pkg/api/types.go

```diff
@@ -1192,13 +1195,15 @@ type ServiceSpec struct {
 	// None can be specified for headless services when proxying is not required
 	ClusterIP string `json:"clusterIP,omitempty"`
 
-	// Type determines how the service will be exposed.  Valid options: ClusterIP, NodePort, LoadBalancer
-	Type ServiceType `json:"type,omitempty"`
 
 	// ExternalIPs are used by external load balancers, or can be set by
 	// users to handle external traffic that arrives at a node.
 	ExternalIPs []string `json:"externalIPs,omitempty"`
 
+	// Only applies to Service Type: LoadBalancer
+	// LoadBalancer will get created with the IP specified in this field.
+	LoadBalancerIP string `json:"loadBalancerIP,omitempty"`
```

Comment:
Just to be clear - there's not a way in OS to say "Allocate me an IP that is mine until I release it" and then say "use that IP to attach a load-balancer (or other ingress mode)?"

@anguslees How does a user in OS pre-configure DNS for a service that isn't up yet?  Or does it just not handle that at all (just as Kubernetes does not handle it today).

@justinsb It sounds like there is no way to pre-configure external DNS in Amazon either, right?  You have to wait until the ELB is instantiated to learn the CNAME to reference.

If both of those are true, then my initial assertion that this was a good idea might be dead wrong.  I would be willing to put a field in the UI that "several" platforms supported even if not all of them did.  I am NOT willing to put a field in the API if only one platform supports it (even if that happens to be my platform :)

Can you guys please confirm my assertions?